src/darksend.cpp

```diff
@@ -248,10 +248,10 @@ void CDarksendPool::ProcessMessage(CNode* pfrom, std::string& strCommand, CDataS
                 }
             }
 
-            // Allow lowest denom (at max) as a a fee. Normally shouldn't happen though.
-            // TODO: Or do not allow fees at all?
-            if(nValueIn - nValueOut > vecPrivateSendDenominations.back()) {
-                LogPrintf("DSVIN -- fees are too high! fees: %lld, tx=%s", nValueIn - nValueOut, tx.ToString());
+            // There should be no fee in mixing tx
+            CAmount nFee = nValueIn - nValueOut;
+            if(nFee != 0) {
```

Comment:
This is just to make sure that client did everything correctly and won't burn funds (un)intentionally. In some _really old_ versions we allowed fees to be payed in the mixing tx which is not the case anymore and all fees are paid via collaterals.